src/VirtoCommerce.MarketingModule.Data/Caching/PromotionUsageCacheRegion.cs

```diff
@@ -11,26 +13,34 @@ public class PromotionUsageCacheRegion : CancellableCacheRegion<PromotionUsageCa
     {
         private static readonly ConcurrentDictionary<string, CancellationTokenSource> _promotionUsageRegionTokenLookup = new ConcurrentDictionary<string, CancellationTokenSource>();
 
-        public static IChangeToken CreateChangeToken(PromotionUsage usage)
+        public static IChangeToken CreateChangeToken(string[] promotionIds)
```

Comment:
Could we use for PromotionUsageCacheRegion  the exists PromotionsCacheRegion instead of creating own?

Reply:
Reworked caching - added individual caching for promotions and usages, added independent caching for search regions.